cloud/smartos/smartos_image_source.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Adam Števko <adam.stevko@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: smartos_image_source
+short_description: Manage image sources on SmartOS compute nodes.
+description:
+    - Add/delete image source on SmartOS compute nodes.
+version_added: "2.0"
+author: Adam Števko (@xen0l)
+options:
+    url:
+        description:
+            - URL of the image server.
+        required: true
+    type:
+        description:
+            - Specifies the type of the image server.
+        required: false
+        default: imgapi
+        choices: [ "imgapi", "docker", "dsapi" ]
+    state:
+        description:
+            - Set or reset the property value.
+        required: true
+        default: present
+        choices: [ "present", "absent" ]
+'''
+
+EXAMPLES = '''
+# Add datasets.at image source
+smartos_image_source: url=http://datasets.at state=present
+
+# Remove official image source
+smartos_image_source: url=https://images.joyent.com state=absent
+
+# Add docker registry as image source
+smartos_image_source: url=https://index.docker.io type=docker
+'''
+
+SUPPORTED_IMAGE_SOURCE_TYPES = ['imgapi', 'docker', 'dsapi']
+
+
+class ImageSource(object):
+
+    def __init__(self, module):
+        self.module = module
+
+        self.url = module.params['url']
+        self.type = module.params['type']
+        self.state = module.params['state']
+
+    def image_source_exists(self):
+        cmd = [self.module.get_bin_path('imgadm')]
+
+        cmd.append('sources')
+
+        if self.module.check_mode:
+            return True
+
+        (rc, out, _) = self.module.run_command(cmd)
+
+        if rc == 0 and self.url in out:
```

Comment:
So what happen if I try to add as a source http://docker.io while a image source 'http://docker.io.example.org' already exist, wouldn't it it be blocked ( since 'http://docker.io' in 'foo\nhttp://docker.io.example.org\n' would be True ?